File: Ad-Hoc/Python/URI-1218.py
def main():
    global_cases = 1
    while True:
        try:
            to_find = input()

            if global_cases > 1:
                print()

            shoes = input().split()

            flag = False
            total_ocurrences = 0
            male_pair = 0
            female_pair = 0

            for information in shoes:
                if information == to_find:
                    total_ocurrences += 1
                    flag = True

                else:
                    if flag:
                        if information == "M":
                            male_pair += 1
                        elif information == "F":
                            female_pair += 1
                        flag = False

            print("Caso {}:".format(global_cases))
            print("Pares Iguais: {}".format(total_ocurrences))
            print("F: {}".format(female_pair))
            print("M: {}".format(male_pair))

            global_cases += 1

        except Exception:
            break


main()

File: Ad-Hoc/Java/URI-1218.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int globalCase = 1;
		
		while(true) {
			String toFind = br.readLine();
			if(toFind == null) break;
			if(globalCase > 1) System.out.println();
			
			String[] shoes = br.readLine().split(" ");
			
			boolean flag = false;
			int totalOcurrences = 0;
			int malePair = 0;
			int femalePair = 0;
			
			for(String information : shoes) {
				
				if(information.equals(toFind)) {
					totalOcurrences += 1;
					flag = true;
					
				}
				
				else {
					
					if(flag) {
						
						if(information.equals("M")) {
							malePair += 1 ;
						}
						else if(information.equals("F")) {
							femalePair += 1;
						}
						flag = false;
						
					}
					
				}
			}
			
			System.out.printf("Caso %d:%n", globalCase);
			System.out.printf("Pares Iguais: %d%n", totalOcurrences);
			System.out.printf("F: %d%n",femalePair);
			System.out.printf("M: %d%n", malePair);
			
			globalCase += 1;
			
		}
			

	}

}
